apps/analysis/serializers.py

```diff
@@ -17,65 +21,82 @@
 )
 
 
-class AnalysisPillarSerializer(UserResourceSerializer):
-    assignee_name = serializers.CharField(source='assignee.username', read_only=True)
-    analysis_title = serializers.CharField(source='analysis.title', read_only=True)
+class AnlyticalEntriesSerializer(UserResourceSerializer):
+    class Meta:
+        model = AnalyticalStatementEntry
+        fields = ('id', 'order', 'entry')
+        read_only_fields = ('analytical_statement',)
+
+
+class AnalyticalStatementSerializer(
+    RemoveNullFieldsMixin,
+    DynamicFieldsMixin,
+    UserResourceSerializer,
+    NestedCreateMixin,
+    NestedUpdateMixin,
+):
+    analytical_entries = AnlyticalEntriesSerializer(source='analyticalstatemententry_set', many=True, required=False)
 
     class Meta:
-        model = AnalysisPillar
+        model = AnalyticalStatement
         fields = '__all__'
-        read_only_fields = ('analysis',)
+        read_only_fields = ('analysis_pillar',)
 
     def validate(self, data):
-        analysis_id = self.context['view'].kwargs.get('analysis_id', None)
-        if analysis_id:
-            data['analysis_id'] = int(analysis_id)
+        analysis_pillar_id = self.context['view'].kwargs.get('analysis_pillar_id', None)
+        if analysis_pillar_id:
+            data['analysis_pillar_id'] = int(analysis_pillar_id)
+        # Validate the analytical_entries
+        analytical_entries_present = AnalyticalStatementEntry.objects.filter(analytical_statement=self.instance).count()
+        if analytical_entries_present + len(data.get('analyticalstatemententry_set', [])) >= ANALYTICAL_ENTRIES_COUNT:
+            raise serializers.ValidationError(f'Analytical entires count must be less than {ANALYTICAL_ENTRIES_COUNT}')
```

Comment:
This should be enough. As entries all(PUT/POST) and none(PATCH) are send through the serializer.

```python3
entries = data.get('analyticalstatemententry_set')

if entries and len(entries) > settings.ANALYTICAL_ENTRIES_COUNT:
    raise serializers.ValidationError(f'Analytical entires count must be less than {ANALYTICAL_ENTRIES_COUNT}')
```

Reply:
Done

Reply:
Let's use `>` not `>=` as we want to have a count up to 50, not 49.